src/viam/module/module.py

```diff
@@ -161,6 +161,7 @@ async def remove_resource(self, request: RemoveResourceRequest):
                 await resource.stop()
             else:
                 resource.stop()
+        await resource.close()
```

Comment:
what if a resource errors while closing?

Reply:
also could we just put the close call inside server.remove_resource()?

Reply:
Perhaps it should go in `ResourceManager.remove_resource` since we always want to make sure that a resource is closed if we're about to remove it. 

My suggestion would be to make `remove_resource` async, because it gives writers of modules a lot more flexibility in case they need to wait for resource cleanup

Reply:
Note: Would making `remove_resource()` async mean a possible breaking change to anyone using modules?

Reply:
would it? do users call `remove_resource()` directly or is it just internal?

Reply:
There aren't any examples of a user calling `remove_resource()`, but if users are able to call `add_resource()`... I guess it'd be a very specific/weird scenario to call it directly. I'll make the change without worry then!